javatests/arcs/android/crdt/CrdtEntityProtoTest.kt

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.crdt
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.core.crdt.CrdtEntity
+import arcs.core.crdt.CrdtSet
+import arcs.core.crdt.CrdtSingleton
+import arcs.core.crdt.VersionMap
+import arcs.core.data.util.ReferencablePrimitive
+import arcs.core.data.util.toReferencable
+import com.google.common.truth.Truth.assertThat
+import kotlin.test.assertFailsWith
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class CrdtEntityProtoTest {
+
+  @Test
+  fun data_toData_dataPropagated() {
+    assertThat(DUMMY_DATA_PROTO.toData()).isEqualTo(DUMMY_DATA)
```

Comment:
A lot of detail is hidden in these `DUMMY_DATA_PROTO`/`DUMMY_DATA` constants. It's hard to verify what the tests are actually doing unless you go and look at the definitions of these constants. 

This kind of thing in tests is a tradeoff between comprehension and brevity, so it can be hard to know when it's good to factor out data into a constant, and when to inline it instead. I'll send you a link with some advice.

Reply:
Discussed over email. Changes have been implemented.